### PR TITLE
masurca: import from brewsci/science

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by sjackman** [comment]
```
curl: (22) The requested URL returned error: 502 Bad Gateway
Error: Failed to download resource "patchelf"
```
https://circleci.com/gh/brewsci/homebrew-bio/5501
Build failed because Bintray is down. 😢

File: Formula/masurca.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  url "https://github.com/alekseyzimin/masurca/releases/download/v3.3.1/MaSuRCA-3.3.1.tar.gz"
+  sha256 "587d0ee2c6b9fbd3436ca2a9001e19f251b677757fe5e88e7f94a0664231e020"
+
+  depends_on "boost" => :build
+  depends_on "bzip2"
+  depends_on "jellyfish"
+  depends_on :linux
+  depends_on "parallel"
+  depends_on "perl"
+  depends_on "zlib"
```
Comment (sjackman) [inline Formula/masurca.rb]:
```suggestion
  unless OS.mac?
    depends_on "bzip2"
    depends_on "zlib"
  end
```

File: Formula/masurca.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class Masurca < Formula
+  # cite Zimin_2013: "https://doi.org/10.1093/bioinformatics/btt476"
+  desc "Maryland Super-Read Celera Assembler"
+  homepage "https://masurca.blogspot.com/"
+  url "https://github.com/alekseyzimin/masurca/releases/download/v3.3.1/MaSuRCA-3.3.1.tar.gz"
+  sha256 "587d0ee2c6b9fbd3436ca2a9001e19f251b677757fe5e88e7f94a0664231e020"
+
+  depends_on "boost" => :build
+  depends_on "bzip2"
```
Comment (sjackman) [inline Formula/masurca.rb]:
```suggestion
```

**Comment by sjackman** [comment]
```
Error: 2 problems in 1 formula detected
  * Dependency 'bzip2' may be unnecessary as it is provided by macOS; try to build this formula without it.
  * Dependency 'zlib' may be unnecessary as it is provided by macOS; try to build this formula without it.
```
https://circleci.com/gh/brewsci/homebrew-bio/5503

I see this formula is Linux only, but `brew audit` still runs on both OS.

**Comment by sjackman** [comment]
```
g++-5: internal compiler error: Killed (program cc1plus)
```
https://circleci.com/gh/brewsci/homebrew-bio/5506

Linux succeeded last build but failed this build. `g++` exceeded the 4 GB of RAM available on CircleCI.

Try adding…
```ruby
    # Reduce memory usage for CircleCI.
    ENV["MAKEFLAGS"] = "-j4" if ENV["CIRCLECI"]
```

**Comment by jonchang** [comment]
Ah, it's already calling `ENV.deparallelize`. I wonder if masurca is doing its own parallelization under the hood...

**Comment by sjackman** [comment]
Yes, it is. Here's the line to patch:
https://github.com/alekseyzimin/masurca/blob/0ce41a5348636a881a380bc2b7683b223e3c0ab0/install.sh.in#L16
https://github.com/alekseyzimin/masurca/blob/0ce41a5348636a881a380bc2b7683b223e3c0ab0/install.sh.in#L20

I suggest…
```ruby
inreplace "install.sh", "make -j $NUM_THREADS", "make -j4"
```

**Comment by jonchang** [comment]
Thanks for the help @sjackman

**Comment by sjackman** [comment]
No worries at all. Thanks for migrating this formula!

**Pull request closed without merging**